### PR TITLE
Homework Browsers/week 1 Caglar

### PR DESCRIPTION
**Pull request by caglar-kaya**

### CODE REVIEW

File: 2-Browsers/Week1/homework/ex4-whatsTheTime/index.js [hunk 1]
```diff
@@ -7,8 +7,20 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
   second). Use `setInterval()` to make sure the time stays current.
 2. Have the function execute when it's loading in the browser.
 ------------------------------------------------------------------------------*/
+setInterval(addCurrentTime, 1000);
+
+const clock = document.createElement('div');
```
Comment (wilgert) [inline 2-Browsers/Week1/homework/ex4-whatsTheTime/index.js:12]: The idea of using `DOMContentLoaded` is that you do not start to access/modify the DOM until it finished loading.
By putting this code outside the `addCurrentTime` function you are actually changing the DOM too early.

Can you come up with a design that does not have this issue? Be warned: do not make the common mistake of calling `setInterval` multiple times ;-)

File: 2-Browsers/Week1/homework/ex3-hijackLogo.js [hunk 2]
```diff
@@ -8,7 +8,9 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
    HackYourFuture logo instead.
 ------------------------------------------------------------------------------*/
 function hijackGoogleLogo() {
-  // TODO your code goes in here
+  const googleLogo = document.querySelector('img[alt="Google"]');
```
Comment (wilgert) [inline 2-Browsers/Week1/homework/ex3-hijackLogo.js:11]: Great job!

File: 2-Browsers/Week1/homework/ex5-catWalk/index.js [hunk 3]
```diff
@@ -21,8 +21,28 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
...
+    if (move >= (innerWidth - catImage.width)) {
+      move = 0;
+    }
+  }
 }
 
-// TODO execute `catWalk` when the browser has completed loading the page
+window.addEventListener('DOMContentLoaded', catWalk);
+
+const mover = setInterval(catWalk, 50);
```
Comment (wilgert) [inline 2-Browsers/Week1/homework/ex5-catWalk/index.js:48]: Your code looks quite nice. Unfortunately your cat gets stuck in the middle of the screen after dancing.
Can you make it move until the end of the screen, and start again at the begin as described in points 5 and 6?

**Review by wilgert — CHANGES_REQUESTED** [review]
You did well. Some small improvements possible on most exercises.
Unfortunately ex5-catWalk is not completely working yet, see comment in that file.

File: 2-Browsers/Week1/homework/ex4-whatsTheTime/index.js (hunk 1 above)
Comment (caglar-kaya) [reply 2-Browsers/Week1/homework/ex4-whatsTheTime/index.js:12]: Hey @wilgert,
First of all, thank you so much for your review my homework and your comments..
I got what you mean. I'll fix this bug as soon as possible..

File: 2-Browsers/Week1/homework/ex3-hijackLogo.js (hunk 2 above)
Comment (caglar-kaya) [reply 2-Browsers/Week1/homework/ex3-hijackLogo.js:11]: Thanks you @wilgert .. :)

File: 2-Browsers/Week1/homework/ex5-catWalk/index.js (hunk 3 above)
Comment (caglar-kaya) [reply 2-Browsers/Week1/homework/ex5-catWalk/index.js:48]: I have tried and also searched on google to make my cat move until the end of the screen but unfortunately I am stuck also like my cat..But I'll rethink about it..

**Comment by caglar-kaya** [comment]
Hi @wilgert ,
I have updated ex4 and ex5..I have inspired from Alex's code on Slack for ex4 and also I'm not fully satisfied with ex5.

File: 2-Browsers/Week1/homework/ex4-whatsTheTime/index.js [hunk 4]
```diff
@@ -7,8 +10,24 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
...
+  clock.style.margin = '40vh auto';
+  clock.style.border = '10px solid red';
+  clock.style.borderRadius = '5px';
+  clock.style.textAlign = 'center';
+  clock.style.width = '400px';
+  clock.style.backgroundColor = '#000';
+  clock.style.color = '#fff';
+  clock.style.fontSize = '6rem';
+  document.body.appendChild(clock);
+  const addCurrentTime = () => {
```
Comment (wilgert) [inline 2-Browsers/Week1/homework/ex4-whatsTheTime/index.js:25]: I would not use the same name as the outer function. It's not clear which one will be called on line 30.

File: 2-Browsers/Week1/homework/ex5-catWalk/index.js [hunk 5]
```diff
@@ -21,8 +21,37 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
...
+    catImage.style.left = move + 'px';
+  }
+
+  const mover = () => {
+    if (move < ((innerWidth - catImage.width) / 2)) {
+      moveCat();
+    }
+    if ((move > ((innerWidth - catImage.width) / 2)) && (move < ((innerWidth + catImage.width) / 2))) {
+      catImage.src = 'https://media1.tenor.com/images/2de63e950fb254920054f9bd081e8157/tenor.gif';
+      setTimeout(() => { move += 10 }, 5000);
```
Comment (wilgert) [inline 2-Browsers/Week1/homework/ex5-catWalk/index.js:45]: How many times do you think this `setTimeout` is called?

File: 2-Browsers/Week1/homework/ex4-whatsTheTime/index.js (hunk 4 above)
Comment (caglar-kaya) [reply 2-Browsers/Week1/homework/ex4-whatsTheTime/index.js:25]: I didn't realise that the names were same. Now I changed inner one as `addUpdatedTime`.

File: 2-Browsers/Week1/homework/ex5-catWalk/index.js (hunk 5 above)
Comment (caglar-kaya) [reply 2-Browsers/Week1/homework/ex5-catWalk/index.js:45]: I did like this:
```
      setTimeout(() => { move += 10; }, 5000);
      counter++;
      console.log(counter);
```
and result was 131 on the console.

Comment (robvk) [reply 2-Browsers/Week1/homework/ex5-catWalk/index.js:49]: @caglar-kaya , that's a little much, should only be called once. Maybe you can think of a way to check if you have already set this in motion and then not do this part?

**Comment by caglar-kaya** [comment]
Hello @robvk 
I have edited setTimeout function and some conditions.

**Pull request closed without merging**